data-prepper-api/src/main/java/org/opensearch/dataprepper/model/configuration/PluginModel.java

```diff
@@ -36,7 +47,7 @@
 @JsonSerialize(using = PluginModel.PluginModelSerializer.class)
 @JsonDeserialize(using = PluginModel.PluginModelDeserializer.class)
 public class PluginModel {
-
+    static final String DEFAULT_PLUGINS_CLASSPATH = "org.opensearch.dataprepper.plugins";
```

Comment:
I don't think this approach is the right one for a few reasons:

1. This is leaking information on the plugins classpath out of core and into Data Prepper API.
2. This class deals with basic serialization and this approach is coupling it with the plugin framework to some degree.
3. This duplicates concerns and code from the plugin framework.
4. The plugin framework supports loading plugins from other packages as well, and this does not honor that.

This class - `PluginModel` - should remain concerned only with the most basic form of a plugin.

Use the plugin framework to detect this metadata. Then, have the schema converter use the plugin framework to get the metadata out of there. It may necessitate new method method on `PluginFactory` (though I'm not sure on this yet).